src/docker/ContextLoadingClient/ContextLoadingClient.ts

```diff
@@ -0,0 +1,167 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from 'vscode';
+import { IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { DockerInfo, PruneResult } from '../Common';
+import { DockerContainer, DockerContainerInspection } from '../Containers';
+import { DockerApiClient, DockerExecCommandProvider, DockerExecOptions } from '../DockerApiClient';
+import { DockerImage, DockerImageInspection } from '../Images';
+import { DockerNetwork, DockerNetworkInspection, DriverType } from '../Networks';
+import { DockerVersion } from '../Version';
+import { DockerVolume, DockerVolumeInspection } from '../Volumes';
+
+export class ContextLoadingClient implements DockerApiClient {
+    private readonly contextLoadingPromise: Promise<void>;
+
+    public constructor() {
+        this.contextLoadingPromise = new Promise((resolve) => {
+            const disposable = ext.dockerContextManager.onContextChanged(() => {
+                disposable.dispose();
+                resolve();
+            });
+        });
+    }
+
+    public dispose(): void {
+        // Do nothing
+    }
+
+    public async info(context: IActionContext, token?: CancellationToken): Promise<DockerInfo> {
+        await this.contextLoadingPromise;

```

Comment:
These two lines are (almost) copied a dozen times; is there a shorthand for this?